libs/client-api/src/v2/README.md

```diff
@@ -0,0 +1,929 @@
+
+2. **Manifest Messages**: When a client sends its first Manifest during initial sync, it may use Rid::default() when it
+   has no history:
+   ```rust
+   // In client_api/src/v2/actor.rs
+   let manifest = ClientMessage::Manifest {
+     object_id,
+     collab_type: CollabType::Unknown,
+     last_message_id: last_message_id.unwrap_or_default(),
+     state_vector: sv.encode_v1(),
+   };
+   ```
+
+3. **Server-side Origin Marking**: The server uses Rid to track message origins:
+   ```rust
+   // In services/appflowy-collaborate/src/group/group_init.rs
+   if let ActionSource::Remote(rid) = source {
+     if rid > current_last_message_id {
+       // Update the last message id
+       update_last_message_id(rid);
+     }
+   }
+   ```
+
+4. **Conflict Resolution**: When handling conflicts during reconnection, the system uses Rid comparisons to determine
+   which update takes precedence:
+   ```rust
+   // In client_api/src/v2/actor.rs
+   last_message_id.rcu(|old| {
+     if rid > **old {
+       Arc::new(rid)
+     } else {
+       old.clone()
+     }
+   });
+   ```
+
+#### Rid in Server-Side Implementation
+
+In the server implementation (`services/appflowy-collaborate`), Rids play a crucial role:
+
+1. **Update Ordering**: The `GroupCommandRunner` in the server uses Rids to process updates in the correct order:
+   ```rust
+   // Pseudocode based on services/appflowy-collaborate/src/group/group_init.rs
+   async fn handle_update(
+     state: &CollabGroupState,
+     origin: &CollabOrigin,
+     update: Vec<u8>,
+     rid: Rid,
+   ) -> Result<(), RTProtocolError> {
+     // Process update
+     // Track the Rid for this update in the server state
+     state.update_last_message_id(rid);
+     // Forward update to other clients with this Rid
+     broadcast_to_subscribers(update, rid);
+   }
+   ```
+
+2. **Update Validation**: The server verifies if an update is already applied using Rid comparison:
+   ```rust
+   // Simplified implementation
+   if received_rid <= last_processed_rid {
+     // Update already processed, can be ignored
+     return Ok(());
+   }
+   ```
+
+3. **Message Forwarding**: When broadcasting updates to other clients, the server includes the original Rid:
+   ```rust
+   // In services/appflowy-collaborate/src/group/group_init.rs
+   let message = BroadcastSync::new(
+     sender,
+     object_id.to_string(),
+     payload,
+     seq_num,
+     original_rid
+   );
+   ```
+
+#### Benefits of Rid-based Ordering
+
+- **Global Ordering**: Provides total ordering of updates across all clients
+- **Idempotency**: Duplicate update detection by tracking last_message_id
+- **Out-of-order Handling**: Later updates are recognized even if they arrive first
+- **Reconnection Efficiency**: Clients request updates newer than their last_message_id
+- **Causality Preservation**: Updates are applied in a causally consistent order
+- **Conflict Resolution Support**: Rid ordering helps resolve conflicts when merged
```

Comment:
The AI really went overboard here ;) 1st point is saying about total ordering, while 5ft is going on about causal order. Pretty much none of these points has any sense.